modules/internal/String.chpl

```diff
@@ -1587,6 +1587,32 @@ module String {
       return _join(S);
     }
 
+    /*
+      :args listToJoin: list contaning elements whose string values will be concatenated
+      .. code-block:: chapel
+      var x = "|".join([1,2,3])
+      writeln(x); //prints "a|10|d"
+    
+    */
+
+    proc join(listToJoin: list) : string {
+      return _join(listToJoin);
+    }
+
+    pragma "no doc"
+    proc _join(listToJoin: list) : string where isArray(listToJoin) {
```

Comment:
This overload will never get called! The where clause restricts it to a type that `isArray`, but list is not an array.